enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,532 @@
+---
+title: core-nework-policies
+authors:
+  - "@bbennett"
+reviewers:
+  - TBD
+approvers: 
+  - TBD
+api-approvers:
+  - No new API
+tracking-link:
+  - "https://issues.redhat.com/browse/OCPSTRAT-819"
+creation-date: 2024-08-13
+last-updated: 2024-11-14
+---
+
+# Network Policies for OpenShift Components
+
+
+## Summary
+
+Kubernetes NetworkPolicy objects allow control over what traffic can
+enter and leave a pod.  Currently, most OpenShift operators do not
+install policies for their namespaces.  This is a vital tool for
+network security that we should use.
+
+## Motivation
+
+OpenShift should use the OpenShift platform capabilities to secure the
+OpenShift platform workloads (including layered products that are
+considered part of the platform).  It should both restrict what can
+talk to our pods, but also what our pods can talk to, so that a
+compromised pod can not be used as a springboard for other attacks.
+
+
+### User Stories
+
+- As an administrator, I want to have confidence that the cluster
+  components are as secure as possible, and are not likely to be an
+  attack vector so that I can trust the platform.
+
+- As an administrator, I need a way to be able to disable the
+  OpenShift policies in case something goes wrong so that I can
+  operate my cluster as it was before while the policies are being
+  fixed
+
+- As an administrator, I need to be able to override specific
+  OpenShift policies to be more restrictive so that I can satisfy my
+  security department
+
+- As an OpenShift release manager, I want to know that all cluster
+  namespaces have policies applied so that I can determine if we are
+  conforming with the policies outlined in this document.
+
+- As an OpenShift support engineer, I need to know if network policy
+  is breaking a service (perhaps not even the service that surfaces
+  the error) before I spend a lot of time debugging things so that I
+  can efficiently diagnose the issue.
+
+
+### Goals
+
+- Make it easy for operators to add network policy support
+
+- Document how to override OpenShift policies
+
+- Audit the traffic to ensure that a policy is not breaking a service
+  and report it in a must-gather
+
+- Audit the policies for compliance
+
+- Ensure all OpenShift namespaces have a default deny-all policy
+  defined, and identify those that do not (including “openshift-”
+  namespaces managed by a non Red Hat operator)
+
+
+### Non-Goals
+
+- We are not considering requiring the definition of Admin Network
+  Policy for components (yet).  But the documentation of the network
+  connections that a namespace uses should make future development of
+  Admin Network Policy easier.  We will be using Admin Network Policy
+  to allow cluster admins to override our namespace Network Policies
+  if needed.
```

Comment:
_why_ is ANP a Non-Goal?

Reply:
We can use ANP as part of the solution, either created by the cluster admin to override, or by our operator with well-known labels that their namespaces can have applied. BUT I don't want to require the components to develop ANP. Partly because I can't work out how to get them added to a cluster... I want the components to manager their own policies and not have a central ANP repositiory. That way the operator can upgrade the policy as needed to ensure that they can seamlessly upgrade. The operator may need to install new policy before an upgrade and then remove the old policy once complete.